### PR TITLE
Mention completion for issue editor.

### PR DESCRIPTION
**Pull request by harryxu**
Mention completion on issue editor.  `@mention` engine from [tribute](https://github.com/zurb/tribute).

### Demo:

New issue:

new issue

View/Edit issue:
view issue

### CODE REVIEW

**Review by tonivj5 — APPROVED** [review]

**Comment by lafriks** [comment]
Can you convert tribute css to less file and include into main gitea css? Then there would be no need for additional css include

File: routers/repo/issue.go [hunk 1]
```diff
@@ -351,6 +351,7 @@ func NewIssue(ctx *context.Context) {
 	ctx.Data["PageIsIssueList"] = true
 	ctx.Data["RequireHighlightJS"] = true
 	ctx.Data["RequireSimpleMDE"] = true
+	ctx.Data["RequireTribute"] = true
```
Comment (ethantkoenig) [inline routers/repo/issue.go:354]: I believe we also should require tribute in [CompareAndPullRequest](https://github.com/go-gitea/gitea/blob/0d4c63fd70d0b255a9a17352cc47ec2b6ca13440/routers/repo/pull.go#L699)

File: templates/base/footer.tmpl [hunk 2]
```diff
@@ -58,6 +58,33 @@
 {{if .RequireDropzone}}
 	<script src="{{AppSubUrl}}/vendor/plugins/dropzone/dropzone.js"></script>
 {{end}}
+
```
Comment (ethantkoenig) [inline templates/base/footer.tmpl]: nit: delete blank line for consistency

File: templates/base/footer.tmpl [hunk 3]
```diff
@@ -58,6 +58,33 @@
...
+		var issuesTribute = new Tribute({
+			values: [
+			{{ range .Assignees }}
+			{key: '{{.Name}} {{.FullName}}', value: '{{.Name}}',
+			name: '{{.Name}}', fullname: '{{.FullName}}', avatar: '{{.RelAvatarLink}}'},
+			{{ end }}
+			],
+			noMatchTemplate: function () { return null },
+			menuItemTemplate: function (item) {
+			var user = item.original;
```
Comment (ethantkoenig) [inline templates/base/footer.tmpl]: nit: should the function body (lines 76-81) be indented?

**Review by ethantkoenig — CHANGES_REQUESTED** [review]
Mostly looks good, just a few minor things

**Comment by harryxu** [comment]
@lafriks I will check out later, but I see other JS plugin's css also included by this way.

**Comment by harryxu** [comment]
@ethantkoenig Done.

**Comment by lafriks** [comment]
@harryxu yes I know, this one has small css so it was just an idea, it's totally not required and not blocking this PR

**Review by ethantkoenig — APPROVED** [review]

**Comment by ethantkoenig** [comment]
LGTM

**Comment by lunny** [comment]
LGTM

**Pull request merged**